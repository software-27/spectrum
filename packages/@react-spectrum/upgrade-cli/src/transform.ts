import path from 'node:path';
import {run as jscodeshift} from 'jscodeshift/src/Runner.js';

const transformPath = path.join(__dirname, 'codemods', 'codemod.js');

interface Options {
  componentSubset?: string,
  path?: string,
  ignorePattern?: string,
  parser?: string
}

export async function transform(options: Options) {
  let {
    path: filePath = '.',
    ...rest
  } = options;

  return await jscodeshift(transformPath, [filePath], rest);
}
