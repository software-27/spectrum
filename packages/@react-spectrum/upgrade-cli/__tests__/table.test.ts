// @ts-ignore
import {defineSnapshotTest} from 'jscodeshift/dist/testUtils';
import transform from '../src/codemods/codemod';

const test = (name: string, input: string) => {
  defineSnapshotTest(transform, {}, input, name);
};

test('Renames variants', `
// import {TableView, TableHeader, TableBody, Column, Row, Cell} from '@adobe/react-spectrum';

<div>
</div>
`);
