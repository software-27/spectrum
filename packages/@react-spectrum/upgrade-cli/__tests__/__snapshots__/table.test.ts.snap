// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renames variants 1`] = `
"// import {TableView, TableHeader, TableBody, Column, Row, Cell} from '@adobe/react-spectrum';

<div>
</div>"
`;
